src/features/blog/containers/newsContainer/index.js

```diff
@@ -10,75 +10,68 @@ import {
   Row,
   GreyWrapper,
 } from '../../../../components/grid'
-// import { ShowMoreButton } from './styles'
 import Button from '../../../../components/button'
 
 const initialPaginationState = {
   end: 9,
   step: 9,
 }
-class NewsContainer extends Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      selectedFilter: props.categories.find(
-        ({ title }) => title === 'All Articles'
-      ),
-      ...initialPaginationState,
-      total: props.articles,
-    }
-  }
+const NewsContainer = ({ articles, categories }) => {
+  const [selectedFilter, setSelectedFilter] = useState(
+    categories.find(({ title }) => title === 'All Articles')
+  )
+  const [initialPagination, setInitialPagination] = useState(
+    initialPaginationState
+  )
+  const [totalArticles, setTotalArticles] = useState(articles)
 
-  handleFilterClick = label => {
-    this.setState((_state, props) => ({
-      selectedFilter: label,
-      ...initialPaginationState,
-      total: props.articles.filter(
+  const handleFilterClick = label => {
+    setSelectedFilter(label)
+    setInitialPagination(initialPaginationState)
```

Comment:
The calling of this function is unnecessary due to the way that `useState` only modifies the state value of the requested property (as opposed to `setState` which modifies the entire state object).

Reply:
Ignore this. We actually do want to call this function to reset the pagination upon every filter select.